driver/others/memory.c

```diff
@@ -447,11 +498,40 @@ static pthread_spinlock_t alloc_lock = 0;
 static BLASULONG  alloc_lock = 0UL;
 #endif
 
+static __inline struct alloc_t ** get_memory_table() {
+#if defined(SMP) && !defined(USE_OPENMP)
```

Comment:
order these to deal with positive cases first? so that we can go if if if/else/else/else ?